src/actions-util.ts

```diff
@@ -875,3 +876,34 @@ export async function isAnalyzingDefaultBranch(): Promise<boolean> {
 export function sanitizeArifactName(name: string): string {
   return name.replace(/[^a-zA-Z0-9_\\-]+/g, "");
 }
+
+// REVIEW: Not sure this is the best place for this function?
+export function doesDirectoryExist(dirPath: string) {
+  try {
+    const stats = fs.lstatSync(dirPath);
+    return stats.isDirectory();
+  } catch (e) {
+    return false;
+  }
+}
+
+export async function uploadDebugArtifacts(
+  toUpload: string[],
+  rootDir: string,
+  artifactName: string
+) {
+  if (toUpload.length === 0) {
+    return;
+  }
+  let suffix = "";
+  const matrix = getRequiredInput("matrix");
+  if (matrix !== undefined && matrix !== "null") {
```

Comment:
Is `matrix === ''` ok? Probably not. This is more succinct and likely more correct:

```suggestion
  if (matrix) {
```